Extensions/permalink_copy.js

```diff
@@ -0,0 +1,124 @@
+//* TITLE permalink_copy **//
+//* VERSION 1.0.0 **//
+//* DESCRIPTION **//
+//* DEVELOPER New-XKit **//
+//* FRAME false **//
+//* BETA false **//
+
+
+let lastTarget;
+
+const copyArea = document.createElement("input");
+copyArea.type = "text";
+copyArea.style = "position: absolute; top: -99em; left:0;";
+
+/*
+ * only run for links of type permalink. LMB: Hook this way to ensure browser compatibility. 
+ */
+function shouldRun(target) {
+	return (target && target.nodeName === "A" && target.classList.contains('permalink'));
+}
+
+function onPermalinkClick(event) {
+
+	if (!shouldRun(event.target)) return;
+
+	const swallowNewWindow = "auxclick" !== event.type;
+	const prefs = XKit.extensions.permalink_copy.preferences;
+	let preference = swallowNewWindow ? prefs.activate : prefs.activate_mmb;
+	if (!!preference && !preference.value) {
+		return;
+	}
+
+	// prevent button from opening a new tab
+	preference = prefs.no_prevent_default;
+
+	if (swallowNewWindow && !(preference && preference.value)) {
+		event.preventDefault();
+	}
+
+	lastTarget = event.target;
+
+	document.body.appendChild(copyArea);
+	const link = lastTarget.href;
+	copyArea.value = link;
+	copyArea.select();
+	const success = document.execCommand('Copy');
+	
+	if (XKit.extensions.permalink_copy.preferences.show_toast.value) {
+		XKit.notifications.add(success ? "permalink copied to clipboard" : "couldn't copy link");
+	}	
+	document.body.removeChild(copyArea);
+	return false;
+}
+
+
+
+function interceptClickEvent(event) {	 
+	if (shouldRun(event.target)) onPermalinkClick(event);
+}
+
+function interceptMmbClickEvent(event) { 
+	if (shouldRun(event.target)) onPermalinkClick(event);
+}
+
+XKit.extensions.permalink_copy = new Object({
+	
+	running: false,
+	preferences: {
+		"sep0": {
+			text: "Activate",
+			type: "separator"
+		},
+		"activate": {
+			text: "Left click on link: Activate",
+			default: true,
+			value: true,
+			desktop_only: true
+		},
+		"activate_mmb": {
+			text: "Mousewheel click: Activate",
+			default: true,
+			value: true,
+			desktop_only: true
+		},
+		"sep1": {
+			text: "New window",
+			type: "separator"
+		},
+		"no_prevent_default": {
+			text: "Do not 'eat up' new window on left-click<br> Did you know? MMB click on a link will open it in a new window or tab!",
+			default: false,
+			value: false,
+			desktop_only: true
+		},
+		"sep2": {
+			text: "Notifications",
+			type: "separator"
+		},
+		"show_toast" : {
+			type: "checkbox",
+			text: "Show notification?",
+			default: true,
+			value: true,
+			desktop_only: true
+		}
+	},
+	
+	
+	run: function() {
```

Comment:
Move this to the top as well, under the preferences, so that the internal functions are between the `run` and `destroy` functions for consistency

Reply:
Can you rephrase? I didn't register any object functions to save two stackframes. Want them inside of it?

Reply:
Oh sorry, I misread the code a bit. Like this is fine to me :stuck_out_tongue: